### ISSUE TITLE
Remove all video/audio tracks when merging files

### ISSUE THREAD

**Issue by Infinitay** [mifi/lossless-cut#634]
**Is your feature request related to a problem? Please describe.**
I found out that if your files have different durations, the resulting merge will have pauses before going to the next file. It makes the merged video look as if the video freezes during transitions. However, if the files you want merged **does not have audio tracks to begin with**, this problem does not happen. I tested it by merging the files when I exported the segments with all audio tracks and without. The exported segments that discarded the audio tracks had no problem and played seamlessly.

**Describe the solution you'd like**
Give the use an option if they would like to discard all audio tracks. If this were to be implemented, I would suggest also adding the ability to remove all video tracks in order to provide the option should it be needed in the future.

**Describe alternatives you've considered**
The alternative is to _manually_ remove the audio tracks from all the files you want merged, and then use the merge tool.

**Additional context**

**Comment by markusd1984** [comment]
So if you have recordings that where split into multiple files they can't be exactly merged together? Wonder if this requires a normal cut mode to get the keep the remaining frames from the ending of video files for exact transition to the next video file?

If the issue can be fixed it would be more helpful then deleting audio/video tracks.

**Comment by Infinitay** [comment]
> So if you have recordings that where split into multiple files they can't be exactly merged together

No, the different segments can be merged. The issue with the merge, as I stated, is that if there is an audio track, and if the durations of the clips to be merged differ, there will be _pauses_/_freezes_ in the merged clip between the merged clips.

I suppose the issue is with ffmpeg itself and the [concat ](https://ffmpeg.org/ffmpeg-formats.html#concat)functionality. However, apparently this could be mitigated/resolved by using [inpoint/outpoints](https://ffmpeg.org/ffmpeg-formats.html#concat) within the text file containing the files you want concatenated. That being said, when [I looked over the code](https://github.com/mifi/lossless-cut/blob/ca0b9d58918b7864f3e7d4978dce01e79456f114/src/ffmpeg.js#L493), we keep it simple and just concat the files. So this **possible** suitable _fix_ isn't ideal or rather the most efficient way to fix things.

I think the most efficient way to solve this would simply be as I suggested: give an option to remove the audio tracks before merging, merge the temporary files created as a result of the no-audio merge, then clean-up those temporary files. Same thing with respective to no video as opposed to no audio.

---

Again, I would like to point out that I don't believe this is necessarily an issue with ffmpeg's concat because **supposedly** it could be mitigated using inpoint/output and/or duration when using the concat command. However, LossLess Cut does not do this.

**Comment by markusd1984** [comment]
I didn't say they can't be merged, emphasis was on not exactly merged as in smooth transitioning from the last frame of the first video unto the first frame of the 2nd video... (I understand we are talking about the dropping multiple files into LC to merge them).

I'm not following you how you practically would use the merged file with audio track missing, unless if your suggesting a workflow workaround to merge separately video only and audio only and then combine the merged video with the merged audio?!? Certainly overkill if we could solve the main issue instead.

My understanding is that in keyframe cut it will drop some frames and thus would expect the same may happen during the merging, not sure if concat has anything to do with it but @mifi knows probably.

**Comment by Infinitay** [comment]
> I didn't say they can't be merged

My bad, I assumed you meant such when you questioned it in the initial. I feel like we are misunderstanding each other and talking about two separate issues lol.

> if what you're suggesting is a workflow workaround to merge separately video only and audio only and then combine the merged video with the merged audio?!?

Simply removing the audio tracks and merging those clips solves the issue. The reason I also suggested to remove video tracks too was simply to have that option should anyone ever need it such as if they needed to only merge audio clips.

> I would suggest also adding the ability to remove all video tracks **_in order to provide the option should it be needed in the future_**

**Comment by markusd1984** [comment]
no problem :) I was mainly focused at **video merging that includes audio**, as I would expect that to be the most common video merging people would want to do using LC, thus slightly confused why we are not looking at the main issue at hand, whatever causes the freeze or gaps between the transition. 😄

I understood you that removing the audio solved your issue, which is interesting indeed, but how will you get the audio back from both videos into the merged video without audio? Don't you ultimately need this?

FYI - Just noticed in the new 3.31.1 version the merge dialogue change from `Sort your files for merge` to `Merge/concatenate files` with more info: 
> This dialog can be used to concatenate files in series, e.g. one after the other:
> [file1][file2][file3]
> It can NOT be used for merging tracks in parallell (like adding an audio track to a video).
> Make sure all files are of the exact same codecs & codec parameters (fps, resolution etc).

Have you compared both video file parameters e.g. with MediaInfo that only duration is the only difference?
(I wonder if when editing the first video in LC using keyframe cut and remove the last few frames before the previous keyframe and merging the export with the second video doesn't show this issues?)

**Comment by mifi** [comment]
Thanks for doing research on this. I think this is a very hard problem to figure out that depends on a lot of parameters such as the codecs, format of the file as well as cutpoints being before/after keyframe, and cut mode in question as well as ffmpeg's implementation for the particular combination. I've seen this happen for other people too, but for some it does not happen. Unfortunately I don't have time or expertise to look into these kind of issues, but hopefully in future versions of ffmpeg it could get improved.
